src/lib/enr/waku2_codec.ts

```diff
@@ -0,0 +1,50 @@
+import { WAKU2_FIELD_LENGTH } from "./constants";
+
+export interface Waku2 {
+  relay: boolean;
+  store: boolean;
+  filter: boolean;
+  lightpush: boolean;
+}
+
+enum Waku2FieldPositions {
+  RELAY = 0,
+  STORE = 1,
+  FILTER = 2,
+  LIGHTPUSH = 3,
+}
+
+const protocols = ["relay", "store", "filter", "lightpush"];
```

Comment:
I find it confusing to have a lower case variable name for a var that is defined in the scope of the whole file.
Also, the order matters here, what do you think of:
```suggestion
const ProtocolsInOrder = ["relay", "store", "filter", "lightpush"];
```